salt/client/__init__.py

```diff
@@ -567,10 +567,14 @@ def cmd_batch(
         if 'gather_job_timeout' in kwargs:
             opts['gather_job_timeout'] = kwargs['gather_job_timeout']
 
+        eauth=kwargs.get('eauth', {})
```

Comment:
Thanks. We're nearly there! Those values won't always be present so pop'ing without a default value will throw a `KeyError`.

If you want to try this at the Python REPL, typing the lines below should do the trick. The three scenarios we care about are: eauth with creds, eauth with an existing token, no eauth.

```py
import salt.client
local = salt.client.LocalClient()
local.cmd_batch('*', 'test.ping', username='<username>', password='<password>', eauth='<eauth backend>')
local.cmd_batch('*', 'test.ping', token='<eauth token>')
local.cmd_batch('*', 'test.ping')
```